From 76b403412003c228afb0222ea68c53420b8016cf Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Tue, 1 Dec 2015 13:02:05 +1100
Subject: [PATCH 5/6] dpnet: Implement IDirectPlay8Client Connect

WIP.
- Connect to Server
- Send INTERNAL msg DN_INTERNAL_MESSAGE_PLAYER_CONNECT_INFO or
                    DN_INTERNAL_MESSAGE_PLAYER_CONNECT_INFO_EX
- Revice DN_SEND_CONNECT_INFO.
- Send DN_ACK_CONNECT_INFO
- Send message to caller.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/dpnet/client.c | 45 +++++++++++++++++++++++++++++++++++++--------
 1 file changed, 37 insertions(+), 8 deletions(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index a066f46..0306d12 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -419,15 +419,44 @@ static HRESULT WINAPI IDirectPlay8ClientImpl_CancelAsyncOperation(IDirectPlay8Cl
 }
 
 static HRESULT WINAPI IDirectPlay8ClientImpl_Connect(IDirectPlay8Client *iface,
-        const DPN_APPLICATION_DESC * const pdnAppDesc, IDirectPlay8Address * const pHostAddr,
-        IDirectPlay8Address * const pDeviceInfo, const DPN_SECURITY_DESC * const pdnSecurity,
-        const DPN_SECURITY_CREDENTIALS * const pdnCredentials, const void * const pvUserConnectData,
-        const DWORD dwUserConnectDataSize, void * const pvAsyncContext,
-        DPNHANDLE * const phAsyncHandle, const DWORD dwFlags)
+        const DPN_APPLICATION_DESC * const app_desc, IDirectPlay8Address * const host_addr,
+        IDirectPlay8Address * const device, const DPN_SECURITY_DESC * const security,
+        const DPN_SECURITY_CREDENTIALS * const credentials, const void * const user_data,
+        const DWORD data_size, void * const context,
+        DPNHANDLE * const async_handle, const DWORD flags)
 {
-  IDirectPlay8ClientImpl *This = impl_from_IDirectPlay8Client(iface);
-  FIXME("(%p):(%p,%p,%x): Stub\n", This, pvAsyncContext, phAsyncHandle, dwFlags);
-  return DPN_OK; 
+    IDirectPlay8ClientImpl *This = impl_from_IDirectPlay8Client(iface);
+
+    FIXME("(%p) : (%p, %p, %p, %p, %p, %p, %u, %p, %p, %x)\n", This, app_desc, host_addr,
+        device, security, credentials, user_data, data_size, context, async_handle, flags);
+
+    if(!This->msghandler)
+        return DPNERR_UNINITIALIZED;
+
+    // Create thread for data?
+    // Pass through to Application.
+    /*
+       PNMSG_CONNECT_COMPLETE complete;
+       complete.dwSize = sizeof(complete);
+       complete.hAsyncOp = NULL;
+       complete.pvUserContext = NULL;
+       complete.hResultCode = DPN_OK;
+       complete.pvApplicationReplyData = NULL;
+       complete.dwApplicationReplyDataSize = 0;
+complete.dpnidLocal = 0x000000ff; // DPNID
+
+     PDPNMSG_CONNECT_COMPLETE pConnectCompleteMsg;
+            pConnectCompleteMsg = (PDPNMSG_CONNECT_COMPLETE)pMsgBuffer;
+
+            // Set m_hrConnectComplete, then set an event letting
+            // everyone know that the DPN_MSGID_CONNECT_COMPLETE msg
+            // has been handled
+            m_hrConnectComplete = pConnectCompleteMsg->hResultCode;
+     */
+
+    SetEvent(This->enumhosts);
+
+    return DPN_OK;
 }
 
 static HRESULT WINAPI IDirectPlay8ClientImpl_Send(IDirectPlay8Client *iface,
-- 
1.9.1

